### PR TITLE
Fix PropertyVector<T*> for multi-component case

### PR DESCRIPTION
**Pull request by norihiro-w**
Currently `PropertyVector<T*> ` works only with a single component.

### CODE REVIEW

File: MeshLib/PropertyVector.h [hunk 1]
```diff
@@ -185,6 +197,20 @@ friend class Properties;
         return t;
     }
 
+    //! Returns the value for the given component stored in the given tuple.
+    T const& getComponent(std::size_t tuple_index,
+                                      std::size_t component) const
```
Comment (TomFischer) [inline MeshLib/PropertyVector.h]: At least on github the indentation seems to be not correct.

File: MeshLib/PropertyVector.h [hunk 2]
```diff
@@ -155,13 +156,24 @@ friend class Properties;
...
     {
-        _values[group_id] = new T(value);
+        T* p = new T[1];
+        p[0] = value;
+        _values[group_id] = p;
+    }
+
+    void initPropertyValue(std::size_t group_id, std::vector<T> const& values)
+    {
+        T* p = new T[values.size()];
```
Comment (TomFischer) [inline MeshLib/PropertyVector.h]: Maybe a check `values.size() == _n_components` could be inserted?

File: MeshLib/PropertyVector.h [hunk 3]
```diff
@@ -155,13 +156,24 @@ friend class Properties;
 
     void initPropertyValue(std::size_t group_id, T const& value)
     {
-        _values[group_id] = new T(value);
+        T* p = new T[1];
```
Comment (TomFischer) [inline MeshLib/PropertyVector.h]: Maybe a check `_n_components == 1` could be inserted?

**Comment by TomFischer** [comment]
If tests are ok: :+1:

File: MeshLib/PropertyVector.h [hunk 4]
```diff
@@ -155,13 +156,31 @@ friend class Properties;
 
     void initPropertyValue(std::size_t group_id, T const& value)
     {
-        _values[group_id] = new T(value);
+        if (_n_components != 1)
+            OGS_FATAL("Single-component version of initPropertyValue() is called "
+                      "for a multi-components PropertyVector<T*>");
+        T* p = new T[1];
+        p[0] = value;
```
Comment (chleh) [inline MeshLib/PropertyVector.h]: Note: This will not work if `T` is an abstract class or `T` is not copyable. However, it did not work either before.

**Comment by norihiro-w** [comment]
@chleh As far as I know, `PropertyVector<T*>` is used only in tests at the moment. I fully agree with your concerns. It should be discussed at some point.

**Pull request merged**